src/main/webapp/script.js

```diff
@@ -47,22 +51,80 @@ function initializeRow(json, key, currentRow) {
   let index = 0;
   for (const type of json[key]) {
     const columnEntry = document.createElement('td');
-    console.log(type);
-    console.log(type.toUpperCase());
+
     if ((type.toUpperCase() === 'POSE' && index === 0) ||
       (type.toUpperCase() === 'POINTCLOUD' && index === 1)) {
       const link = document.createElement('a');
       link.href = '/getrun?id=' + key + '&dataType=' + json[key];
       link.innerText = type;
       columnEntry.appendChild(link);
+      currentRow.appendChild(columnEntry);
+      /* Makes checkboxes for each data entry. */
+      const checkbox = document.createElement('INPUT');
+      checkbox.type = 'checkbox';
+      checkbox.setAttribute('name', 'check');
+      checkbox.setAttribute('value', link.getAttribute('name'));
+      currentRow.appendChild(checkbox);
     } else {
       columnEntry.innerText = '';
+      currentRow.appendChild(columnEntry);
     }
-    currentRow.appendChild(columnEntry);
     index++;
   }
   return currentRow;
 }
+
+/* Select all Checkboxes function. */
+function toggle(source) {
+  checkboxes = document.getElementsByName('check');
+  for (let i=0, n=checkboxes.length; i<n; i++) {
+    checkboxes[i].checked = source.checked;
+  }
+}
+
+/*
+ * Make url for multiple runs selected. should be in the style of
+ ** link.href = "/home.html?id=" + key + "&dataType=" + json[key]
+ */
+function makeMultiRunURL() {
+  const checkboxes = document.getElementsByName('check');
+  let count =0;
+  const link = document.createElement('a');
+  link.href = '/home.html?';
+
+  for (let i=0, n=checkboxes.length; i<n; i++) {
+    if (checkboxes[i].checked) {
+      count++;
+      console.log('here what we are splitting!' +
+       checkboxes[i].getAttribute('value'));
+      const splitInfo = checkboxes[i].getAttribute('value').split('_');
+      link.href += 'id' + count + '=' + splitInfo[0] + '&type' +
```

Comment:
`+=` on a string within a loop is always a red flag, because the string has to be copied on each iteration. You could accumulate substrings within an array, then use `Array.join` to concatenate them all outside of the loop.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/join